### PR TITLE
🐛 Source Jira: fix bug when `board` doesn't support `sprints`

### PR DESCRIPTION
**Pull request by bazarnov**
## What
Resolving: https://github.com/airbytehq/oncall/issues/2133

## How
- added `should_retry == False` logic to skip the slice with `board` that doesn't support  `sprints` + informative logger message about this
- covered the case with unit test

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

File: airbyte-integrations/connectors/source-jira/source_jira/streams.py [hunk 1]
```diff
@@ -92,6 +93,15 @@ def read_records(self, **kwargs) -> Iterable[Mapping[str, Any]]:
             if not (self.skip_http_status_codes and e.response.status_code in self.skip_http_status_codes):
                 raise e
```
Comment (sherifnada) [inline airbyte-integrations/connectors/source-jira/source_jira/streams.py:95]: As someone with no context on this issue reading the code this looks like we're sweeping errors under the rug. Could you leave a source code comment explaining why we do this?

Comment (bazarnov) [reply airbyte-integrations/connectors/source-jira/source_jira/streams.py:95]: that was no part of the original fix) i'll check and leave the comment what's going on there.

Comment (sherifnada) [reply airbyte-integrations/connectors/source-jira/source_jira/streams.py:95]: I meant the lines you added :)

**Review by sherifnada — APPROVED** [review]

**Review by girarda — CHANGES_REQUESTED** [review]
We should use the an availability strategy and remove the Sprint stream from `source.streams()` if it's not available instead of ignoring errors

**Comment by bazarnov** [comment]
> We should use the an availability strategy and remove the Sprint stream from `source.streams()` if it's not available instead of ignoring errors

I disagree here, we should ignore the slice with board that is not available for sprints, since it's not supported, not the entire stream, because 1 board doesn't support the sprints @girarda

**Review by girarda — APPROVED** [review]
@bazarnov this makes sense. can you add context to the code for future reference? 

Without context,
- the `should_retry` method seems to be blindly ignoring failures
- It's not obvious that we should only skip a single slice and why

File: airbyte-integrations/connectors/source-jira/source_jira/streams.py (hunk 1 above)
Comment (bazarnov) [reply airbyte-integrations/connectors/source-jira/source_jira/streams.py:95]: Done.

**Pull request merged**